Framework/Backend/test/mocha-http.js

```diff
@@ -40,222 +39,129 @@ describe('REST API', () => {
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .expect({body: postData}, done);
   });
 
   it('POST with an incorrect token should respond 403', (done) => {
-    const req = http.request({
-      hostname: 'localhost',
-      port: config.http.port,
-      path: '/api/post-request?token=wrong',
-      method: 'POST'
-    }, (res) => {
-      assert.strictEqual(res.statusCode, 403);
-      done();
-    });
-    req.end();
+    request(httpServer)
+      .post('/api/post-request?token=wrong')
+      .expect('Content-Type', /json/)
+      .expect(403, done);
   });
 
   it('PUT with a token should respond 200/JSON', (done) => {
-    const req = http.request({
-      hostname: 'localhost',
-      port: config.http.port,
-      path: '/api/put-request?token=' + token,
-      method: 'PUT'
-    }, (res) => {
-      assert.strictEqual(res.statusCode, 200);
-      let rawData = '';
-      res.on('data', (chunk) => {
-        rawData += chunk;
-      });
-      res.on('end', () => {
-        const parsedData = JSON.parse(rawData);
-        assert.strictEqual(parsedData.ok, 1);
-        done();
-      });
-    });
-    req.end();
+    request(httpServer)
+      .put('/api/put-request?token=' + token)
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .expect({ok: 1}, done);
   });
 
   it('PATCH with a token should respond 200/JSON', (done) => {
-    const req = http.request({
-      hostname: 'localhost',
-      port: config.http.port,
-      path: '/api/patch-request?token=' + token,
-      method: 'PATCH'
-    }, (res) => {
-      assert.strictEqual(res.statusCode, 200);
-      let rawData = '';
-      res.on('data', (chunk) => {
-        rawData += chunk;
-      });
-      res.on('end', () => {
-        const parsedData = JSON.parse(rawData);
-        assert.strictEqual(parsedData.ok, 1);
-        done();
-      });
-    });
-    req.end();
+    request(httpServer)
+      .patch('/api/patch-request?token=' + token)
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .expect({ok: 1}, done);
   });
 
   it('DELETE with a token should respond 200/JSON', (done) => {
-    const req = http.request({
-      hostname: 'localhost',
-      port: config.http.port,
-      path: '/api/delete-request?token=' + token,
-      method: 'DELETE'
-    }, (res) => {
-      assert.strictEqual(res.statusCode, 200);
-      let rawData = '';
-      res.on('data', (chunk) => {
-        rawData += chunk;
-      });
-      res.on('end', () => {
-        const parsedData = JSON.parse(rawData);
-        assert.strictEqual(parsedData.ok, 1);
-        done();
-      });
-    });
-    req.end();
+    request(httpServer)
+      .delete('/api/delete-request?token=' + token)
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .expect({ok: 1}, done);
   });
 });
 
 describe('HTTP server', () => {
-  after(() => {
-    httpServer.getServer.close();
+  after(async () => {
+    await httpServer.close();
```

Comment:
As we are going with the flag approach, there is no need to for changes in the `after` section